src/sql/src/catalog.rs

```diff
@@ -1328,6 +1330,52 @@ impl<'a, T> ErsatzCatalog<'a, T> {
     }
 }
 
+/// Provides a method of generating a 2-layer catalog on the fly, and then
+/// resolving objects within it.
+pub(crate) struct SubsourceCatalogDouble<'a, T>(pub BTreeMap<String, BTreeMap<String, &'a T>>);
+
+impl<'a, T> SubsourceCatalog<'a, T> for SubsourceCatalogDouble<'a, T> {
+    /// Returns the fully qualified name for `item`, as well as the `T` that it
+    /// describes.
+    ///
+    /// # Errors
+    /// - If `item` cannot be normalized to a [`PartialItemName`]
+    /// - If the normalized `PartialItemName` does not resolve to an item in
+    ///   `self.0`.
+    fn resolve(&self, item: UnresolvedItemName) -> Result<(UnresolvedItemName, &'a T), PlanError> {
+        let name = normalize::unresolved_item_name(item)?;
```

Comment:
Should this fn assert anything about `name.database`?

Reply:
No longer applicable after removing the 2-layer catalog implementation due to comments below